src/plugins/good-meetup-tracking.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+// Load Modules
+
+const avro = require('avsc');
+const Hoek = require('hoek');
+const request = require('request');
+const Stream = require('stream');
+
+const internals = {
+	defaults: {
+		endpoint: 'http://log.analytics.mup-prod.mup.zone/log',
+		// in prod, make a `request` call, otherwise no-op
+		postData: process.env.NODE_ENV === 'production' ?
+			request.post.bind(request) :
+			() => {},
+		// currently the schema is manually copied from
+		// https://github.dev.meetup.com/meetup/meetup/blob/master/modules/base/src/main/versioned_avro/Activity_v3.avsc
+		schema: avro.parse({
+			namespace: 'com.meetup.base.avro',
+			type: 'record',
+			name: 'Activity',
+			doc: 'v3',
+			fields: [
+				{ name: 'requestId', type: 'string'},
+				{ name: 'timestamp', type: 'string'},
+				{ name: 'url', type: 'string'},
+				{ name: 'aggregratedUrl', type: 'string', default: ''},  // it's misspelled in the original spec
+				{ name: 'ip', type: 'string', default: ''},
+				{ name: 'agent', type: 'string', default: ''},
+				{ name: 'memberId', type: 'int'},
+				{ name: 'trackId', type: 'string'},
+				{ name: 'mobileWeb', type: 'boolean'},
+				{ name: 'platform', type: 'string'},
+				{ name: 'referer', type: 'string'},  // it's misspelled in the original spec
+				{ name: 'trax', type: { type: 'map', values: 'string'}},
+				{
+					name: 'platformAgent',
+					type: {
+						type: 'enum',
+						name: 'PlatformAgent',
+						symbols: ['WEB', 'NATIVE', 'NATIVE_APP_WEB_VIEW', 'THIRD_PARTY_UNKNOWN', 'UNKNOWN']
+					},
+					default:'UNKNOWN'
+				}
+			]
+		}),
+	},
+};
+
+/**
+ * A stream that transforms a Good event into an Avro buffer according to a
+ * schema
+ */
+class GoodMeetupTracking extends Stream.Transform {
+	/**
+	 * @param {Object} config the config options for a Stream.Transform
+	 * @return {undefined} side effects only
+	 */
+	constructor(config) {
+		super({ objectMode: true });
+
+		config = config || {};
+		this._settings = Hoek.applyToDefaults(internals.defaults, config);
+	}
+
+	/**
+	 * @param {Error|null} err error returned from failed request
+	 * @param {http.IncomingMessage} response the response object
+	 * @param {String} body the body of the response
+	 * @return {undefined} side effects only
+	 */
+	static postDataCallback(err, response, body) {
+		if (err) {
```

Comment:
I believe that a timeout would populate the `err` value, but a non-200 response would _not_ populate `err` since the `response` is 'successful' even though it indicates a server error.

So I think we do need both checks - one for `err` and one for non-200 response.